### DISCUSSION TITLE
next.js problem compiling - node modules error

### DISCUSSION THREAD

**Discussion by anthonygarced** [vercel/next.js#43091] [Help] [answered]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

    Operating System:
      Platform: win32
      Arch: x64
      Version: Windows 10 Home
    Binaries:
      Node: 19.1.0
      npm: N/A
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 13.0.4
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0

### What browser are you using? (if relevant)

Not relevant, but Chrome

### How are you deploying your application? (if relevant)

installed via npx create-next-app

### Describe the Bug

You can see the error here:
`wait  - compiling /_error (client and server)...
event - compiled client and server successfully in 80 ms (187 modules)
Warning: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem.
TypeError: Cannot read properties of null (reading 'useContext')
    at Object.useContext (C:\Users\garced\Documents\anthonygarced.com\my-app\node_modules\react\cjs\react.development.js:1618:21)
    at Html (webpack-internal:///../../../AppData/Roaming/npm/node_modules/next/dist/pages/_document.js:678:104)
    at renderWithHooks (C:\Users\garced\AppData\Roaming\npm\node_modules\next\node_modules\react-dom\cjs\react-dom-server.browser.development.js:5658:16)
    at renderIndeterminateComponent (C:\Users\garced\AppData\Roaming\npm\node_modules\next\node_modules\react-dom\cjs\react-dom-server.browser.development.js:5731:15)
    at renderElement (C:\Users\garced\AppData\Roaming\npm\node_modules\next\node_modules\react-dom\cjs\react-dom-server.browser.development.js:5946:7)
    at renderNodeDestructiveImpl (C:\Users\garced\AppData\Roaming\npm\node_modules\next\node_modules\react-dom\cjs\react-dom-server.browser.development.js:6104:11)
    at renderNodeDestructive (C:\Users\garced\AppData\Roaming\npm\node_modules\next\node_modules\react-dom\cjs\react-dom-server.browser.development.js:6076:14)
    at finishClassComponent (C:\Users\garced\AppData\Roaming\npm\node_modules\next\node_modules\react-dom\cjs\react-dom-server.browser.development.js:5688:3)
    at renderClassComponent (C:\Users\garced\AppData\Roaming\npm\node_modules\next\node_modules\react-dom\cjs\react-dom-server.browser.development.js:5696:3)
    at renderElement (C:\Users\garced\AppData\Roaming\npm\node_modules\next\node_modules\react-dom\cjs\react-dom-server.browser.development.js:5943:7)
npm\node_modules\next\node_modules\react-dom\cjs\react-dom-server.browser.development.js:6076:14)
    at finishClassComponent (C:\Users\garced\AppData\Roaming\npm\node_modules\next\node_modules\react-dom\cjs\react-dom-server.browser.development.js:5688:3)
    at renderClassComponent (C:\Users\garced\AppData\Roaming\npm\node_modules\next\node_modules\react-dom\cjs\react-dom-server.browser.development.js:5696:3)
    at renderElement (C:\Users\garced\AppData\Roaming\npm\node_modules\next\node_modules\react-dom\cjs\react-dom-server.browser.development.js:5943:7)`

I have checked multiple times, there is no duplicates of any react folder, I reinstalled the next.js variables by installing them multiple times in different folders, and BY default it creates this error every single time.

I even reinstalled node,js and react and it still creates this issue. I also made a post on StackOverflow, but got no response.
I'm lilkely missing something here, but I'd thought I would make an issue here and see if this would help with the issue.

### Expected Behavior

I expected to see the default page when you first installed next.js, which would be how to get started.

Instead, I got a context error.

### Link to reproduction - Issues with a link to complete (but minimal) reproduction code will be addressed faster

https://github.com/anthonygarced/test-example

### To Reproduce

_This should be how you do it here:_

1. Open VSC (Visual Studio Code)
2.. ctrl `, npx install create-next-app
3. This does not matter if you install ESLint or TypeScript, this issue occurs via the node_modules

**Comment by balazsorban44** [comment]
The repo is private, could you make sure we can access the reproduction?

**Comment by anthonygarced** [comment]
Sure thing.  

One problem is: I do have to upload the entirety of the **node_modules** folder. I will figure this out when I get home, but I will public the repository and upload all of the node_modules files. 
(also ignore the closed and reopens, I thought those buttons required confirmation)

**Comment by balazsorban44** [comment]
It's unclear why you would need to upload the `node_modules` folder.

**Comment by anthonygarced** [comment]
I'd feel I would need the node_modules folder because the primary issue is that

I have made the build for you: https://github.com/anthonygarced/test
If you compile it successfully, I'll make the zip file of everything (including the node_modules) to show you. It happens once you compile it and install with the node_modules.

**Comment by icyJoseph** [comment, marked as answer]
Are you using `useContext` within `_document`? Is  `_document` a `class` or is it a functional component?

## Resolution

OP had installed `next` globally.

**Reply by anthonygarced** [reply]
return dispatcher.useContext(Context); is what is happening when it tries to contact \my-app\node_modules\react\cjs\react.development.js:1618:21). The rest seems to try to touch something within my AppData/Roaming folder, (which is just the node modules itself)
 
The code below is the default app page inside of whenever I installed (and I installed it 10 times with new node_modules, all within the same error. I even reinstalled node.js normally).
`
import '../styles/globals.css'
import type { AppProps } from 'next/app'

export default function App({ Component, pageProps }: AppProps) {
  return <Component {...pageProps} />
}
`

It seems it is using _document as a _functional_ component. Could be wrong. Sorry for late reply.

**Reply by icyJoseph** [reply]
That's really weird, why would it try to read from `AppData/Roaming` rather than your local folder. Your app lives here: `C:\Users\garced\Documents\anthonygarced.com\my-app`, so it should use `next` and `react` + `react-dom` from `C:\Users\garced\Documents\anthonygarced.com\my-app/node_modules`, why is it resolving to the `AppData/Roaming` - Perhaps this is some Windows + Webpack behavior I am unaware of...?

As far as I understand, `AppData/Roaming` is where the node_modules cache is stored, but why would it look for dependencies there. How are you running your project?

**Reply by anthonygarced** [reply]
I am running my project via VSC. I use next instead of next run because when I do Next Run it would cause an error. I found that next runs the project.

I'm not sure if that's the correct way to do it, but the only other thing I have that can run this is Live Servers, and then I use run next from the terminal, I don't hit live server.

I'm on Windows 11 22H2, but I don't see anyone else having this issue.
Sorry for late reply.

**Reply by icyJoseph** [reply]
VSC is VSCode? Once you run the project, all you gotta do is `npm run dev` or `yarn dev` or even pnpm works, but you say you do:

`I use next instead of next run because when I do Next Run it would cause an error. I found that next runs the project.`

What do you mean exactly? What commands are you actually running? I am also confused about why you'd need Live Server, for a next app, unless you are using `next export`? Let's step back... could you, without VSCode, just use a terminal:

- `cd` into your projects folder
-  `npx create next-app delete-later` for example, 
-  then cd into `delete-later`
-  and run `npm run dev`, does the project run?

**Reply by anthonygarced** [reply]
Hello. 

The terminal next thing somehow works. Everything is done, with terminal, **the project does infact run**.

"_VSC is VSCode? Once you run the project, all you gotta do is npm run dev or yarn dev or even pnpm works, but you say you do:

I use next instead of next run because when I do Next Run it would cause an error. I found that next runs the project._"
I do next, because when I did next run it provided me with an error. (and I also didn't think of running those commands).

(live server was also just run basic html documents, i did next export and it sometimes broke).

**Reply by icyJoseph** [reply]
The thing is that it looks like you've got `next` installed globally then.

Let's forget about `next export`, live server, and vscode. Were you able to do the steps above?

**Reply by anthonygarced** [reply]
If you're talking about the project running, then yes I was able to get it running.

**Reply by icyJoseph** [reply]
So no more problems? 👍

**Reply by anthonygarced** [reply]
Yes, but still am confused about how this happened and why it read from appdata.

but other than that, no more problems

**Reply by icyJoseph** [reply]
If you do: `npm list -g --depth 0` it will print a list of globally available packages. Is `next` listed there?

**Reply by anthonygarced** [reply]
Yes, specifically:
+-- create-next-app@13.0.4
+-- next@13.0.2
`-- serve@14.0.1

**Reply by icyJoseph** [reply]
Yeah so that's why it kinda wants to work, because it picked up on that one.... It'd recommend running `npm uninstall -g next`, this will remove it globally. You only need it at the location of your project.

Please don't forget to mark as answered, I've updated this answer with the resolution.

**Reply by anthonygarced** [reply]
I have marked it as answer. Perhaps I should also link this discussion page on the StackOverflow problem I made.

Thanks

**Reply by ELMOUDENX** [reply]
same problem but i dont have react installed globally ? what to doo??

**Reply by icyJoseph** [reply]
React is not next. Double check please.

**Reply by winhack777** [reply]
I could not solve the problem of running next.js out of the box for several days. After the command `npm run dev` in the browser, the error constantly appeared:
"TypeError: Cannot read properties of null (reading 'useReducer')".

But the project was launched from Visual Studio Code with Live Server installed. 
After reading your comment, I tried to delete Live Server and try again - it did not work. But then he completely abandoned VSK and launched the project through the Windows terminal.
It worked!!!
Thank you!!!
